.ci/Jenkinsfile

```diff
@@ -0,0 +1,85 @@
+#!/usr/bin/env groovy
+
+@Library('apm@current') _
+
+pipeline {
+  agent none
+  environment {
+    REPO = 'simple-git'
+    BASE_DIR = "src/github.com/elastic/${env.REPO}"
+    YARN_GPG = 'no'
+  }
+  options {
+    timeout(time: 1, unit: 'HOURS')
+    buildDiscarder(logRotator(numToKeepStr: '30', artifactNumToKeepStr: '20'))
+    timestamps()
+    ansiColor('xterm')
+    disableResume()
+    durabilityHint('PERFORMANCE_OPTIMIZED')
+  }
+  parameters {
+    booleanParam(name: 'publish_to_npm', defaultValue: false, description: 'Publish to NPM.')
+  }
+  triggers {
+    issueCommentTrigger('(?i).*(?:jenkins\\W+)?run\\W+(?:the\\W+)?tests(?:\\W+please)?.*')
+  }
+  stages {
+    stage('Checkout') {
+      agent { label 'linux && immutable' }
+      options { skipDefaultCheckout() }
+      steps {
+        deleteDir()
+        gitCheckout(basedir: "${BASE_DIR}", githubNotifyFirstTimeContributor: true)
+        stash allowEmpty: true, name: 'source', useDefaultExcludes: false
+      }
+    }
+    stage('Build Linux') {
+      agent { label 'linux && immutable' }
+      options { skipDefaultCheckout() }
+      steps {
+        deleteDir()
+        unstash 'source'
+        sh(label: 'Build NodeJS', script: './ci/scripts/build.sh')
+        sh(label: 'Build NodeJS', script: 'yarn install --frozen-lockfile')
+        stash allowEmpty: true, name: 'compiledSource', useDefaultExcludes: false
+      }
+    }
+    stage('Unit Tests') {
+      agent { label 'linux && immutable' }
+      options { skipDefaultCheckout() }
+      environment {
+        HOME = "${env.WORKSPACE}"
+      }
+      steps {
+        unstash 'compiledSource'
+        withGithubNotify(context: 'Test', tab: 'tests') {
+          deleteDir()
+          unstash 'source'
+          dir("${BASE_DIR}"){
+            sh(label: 'Unit Tests', script: 'yarn run test')
+          }
+        }
+      }
+    }
+    stage('Deploy') {
+      agent { label 'linux && immutable' }
+      options { skipDefaultCheckout() }
+      when {
+        beforeAgent true
+        allOf {
+          tag pattern: 'v\\d+.*', comparator: 'REGEXP'
+          expression { return params.publish_to_npm }
+        }
+      }
+      steps {
+        withGithubNotify(context: 'Test', tab: 'tests') {
+          deleteDir()
+          unstash 'compiledSource'
+          dir("${BASE_DIR}"){
+            sh(label: 'Upload NodeJS', script: './ci/scripts/upload.sh')
```

Comment:
If the script contains any access to vault then it's required to use the mask/withEnv steps, maybe the `withSecretVault` step might help for this particular use case.